server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -70,36 +69,58 @@ class InteractiveSession(
     val conf = prepareConf(request.conf, request.jars, request.files, request.archives,
       request.pyFiles)
 
-    info(s"Creating LivyClient for sessionId: $id")
-    val builder = new LivyClientBuilder()
-      .setConf("spark.app.name", s"livy-session-$id")
-      .setAll(conf.asJava)
-      .setConf("livy.client.sessionId", id.toString)
-      .setConf(RSCConf.Entry.DRIVER_CLASS.key(), "com.cloudera.livy.repl.ReplDriver")
-      .setURI(new URI("rsc:/"))
+    val builderProperties = mutable.Map[String, String]()
+    builderProperties ++= conf
 
     def mergeConfList(list: Seq[String], key: String): Unit = {
       if (list.nonEmpty) {
-        val newList = (list ++ conf.get(key)).mkString(",")
-        builder.setConf(key, newList)
+        builderProperties.get(key) match {
+          case None =>
+            builderProperties.put(key, list.mkString(","))
+          case Some(oldList) =>
+            val newList = (oldList :: list.toList).mkString(",")
+            builderProperties.put(key, newList)
+        }
       }
     }
 
+    def mergeHiveSiteAndHiveDeps(): Unit = {
+      val sparkFiles = conf.get("spark.files").map(_.split(",")).getOrElse(Array.empty[String])
+      if (livyConf.getBoolean(LivyConf.ENABLE_HIVE_CONTEXT)) {
+        hiveSiteFile(sparkFiles, livyConf) match {
+          case (_, true) =>
+            info("Enable HiveContext because hive-site.xml is found in user request.")
+            mergeConfList(datanucleusJars(livyConf), LivyConf.SPARK_JARS)
+          case (Some(file), false) =>
+            info("Enable HiveContext because hive-site.xml is found under classpath, "
+              + file.getAbsolutePath)
+            mergeConfList(List(file.getAbsolutePath), LivyConf.SPARK_FILES)
+            mergeConfList(datanucleusJars(livyConf), LivyConf.SPARK_JARS)
+          case (None, false) =>
+            warn("Enable HiveContext but no hive-site.xml found under" +
```

Comment:
But I will check hive-site.xml both in classpath and user request. I think it deserves a warning when both places can not find hive-site.xml